hyperspy/io_plugins/bcf.py

```diff
@@ -593,8 +593,14 @@ def _set_sem(self, root):
         self.sem.wd = float(semData.WD)  # in mm
         self.sem.mag = float(semData.Mag)  # in times
         # image/hypermap resolution in um/pixel:
-        self.image.x_res = float(semData.DX)  # in micrometers
-        self.image.y_res = float(semData.DY)  # in micrometers
+        try:
+            self.image.x_res = float(semData.DX)  # in micrometers
+            self.image.y_res = float(semData.DY)  # in micrometers
+            self.units = 'µm'
+        except AttributeError:
+            self.image.x_res = 1.0  # in pixels
+            self.image.y_res = 1.0  # in pixels
+            self.units = 'pix'
```

Comment:
Usually, when the units aren't defined, we use `Undefined` from the `traits.api`.